stellargraph/mapper/mini_batch_node_generators.py

```diff
@@ -0,0 +1,361 @@
+        if len(self.schema.node_types) > 1:
+            raise TypeError(
+                "{}: node generator requires graph with single node type; "
+                "a graph with multiple node types is passed. Stopping.".format(
+                    type(self).__name__
+                )
+            )
+
+        if not clusters:
+            # We are not given graph clusters.
+            # We are going to split the graph into self.k random clusters
+            all_nodes = list(G.nodes())
+            random.shuffle(all_nodes)
+            cluster_size = len(all_nodes) // self.k
+            self.clusters = [
+                all_nodes[i : i + cluster_size]
+                for i in range(0, len(all_nodes), cluster_size)
+            ]
+            if len(self.clusters) > self.k:
+                # for the case that the number of nodes is not exactly divisible by k, we combine
+                # the last cluster with the second last one
+                self.clusters[-2].extend(self.clusters[-1])
+                del self.clusters[-1]
+
+        print(f"Number of clusters {len(self.clusters)}")
+        for i, c in enumerate(self.clusters):
+            print(f"{i} cluster has size {len(c)}")
+
+        # Get the features for the nodes
+        self.features = G.get_feature_for_nodes(self.node_list)
+
+    def flow(self, node_ids, targets=None, name=None):
+        """
+        Creates a generator/sequence object for training, evaluation, or prediction
+        with the supplied node ids and numeric targets.
+
+        Args:
+            node_ids (iterable): an iterable of node ids for the nodes of interest
+                (e.g., training, validation, or test set nodes)
+            targets (2d array, optional): a 2D array of numeric node targets with shape `(len(node_ids),
+                target_size)`
+            name (str, optional): An optional name for the returned generator object.
+
+        Returns:
+            A ClusterNodeSequence object to use with ClusterGCN in Keras
+            methods :meth:`fit_generator`, :meth:`evaluate_generator`, and :meth:`predict_generator`
+
+        """
+        if targets is not None:
+            # Check targets is an iterable
+            if not is_real_iterable(targets):
+                raise TypeError(
+                    "{}: Targets must be an iterable or None".format(
+                        type(self).__name__
+                    )
+                )
+
+            # Check targets correct shape
+            if len(targets) != len(node_ids):
+                raise TypeError(
+                    "{}: Targets must be the same length as node_ids".format(
+                        type(self).__name__
+                    )
+                )
+
+        return ClusterNodeSequence(
+            self.graph,
+            self.clusters,
+            targets=targets,
+            node_ids=node_ids,
+            q=self.q,
+            name=name,
+        )
```

Comment:
Do we not pass `self.lam` into `ClusterNodeSequence`?

Reply:
Fixed!